Paraswap-Agents/delegate-calls/src/agent.spec.ts

```diff
@@ -0,0 +1,153 @@
+
+    // Set the call type of every trace to "delegatecall"
+    tx.traces.forEach((trace) => {
+      trace.action.callType = "delegatecall";
+    });
+
+    const findings: Finding[] = await handler(tx);
+
+    expect(findings).toStrictEqual([]);
+  });
+
+  it("ignores delegated calls to functions other than `megaSwap` and `multiSwap` from AugustusSwapper", async () => {
+    const tx: TransactionEvent = new TestTransactionEvent()
+      .addTraces({
+        to: EXTERNAL_ADDR_1,
+        from: AUGUSTUS_ADDR,
+        // Function signature that does not match `multiSwap` or `megaSwap`
+        input: "0xa1b2c3d4",
+        output: "0x0",
+      });
+
+    // Set the call type of every trace to "delegatecall"
+    tx.traces.forEach((trace) => {
+      trace.action.callType = "delegatecall";
+    });
+
+    const findings: Finding[] = await handler(tx);
+
+    expect(findings).toStrictEqual([]);
+  });
+
+  it("detects delegated calls to `megaSwap` and `multiSwap` from AugustusSwapper", async () => {
+    const tx: TransactionEvent = new TestTransactionEvent()
+      .addTraces({
+        to: EXTERNAL_ADDR_1,
+        from: AUGUSTUS_ADDR,
+        // Manually place the function signature due to function argument complexity
+        input: "0xa94e78ef",
+        output: "0x0",
+      })
+      .addTraces({
+        to: EXTERNAL_ADDR_1,
+        from: AUGUSTUS_ADDR,
+        // Manually place the function signature due to function argument complexity
+        input: "0x46c67b6d",
+        output: "0x0",
+      });
+
+    // Set the call type of every trace to "delegatecall"
+    tx.traces.forEach((trace) => {
+      trace.action.callType = "delegatecall";
+    });
+
+    const findings: Finding[] = await handler(tx);
+
+    expect(findings).toStrictEqual([
+      createFinding(EXTERNAL_ADDR_1, "0xa94e78ef"),
+      createFinding(EXTERNAL_ADDR_1, "0x46c67b6d"),
+    ]);
+  });
+
+  it("detects multiple delegated calls to `megaSwap` and `multiSwap` from AugustusSwapper", async () => {
+    const tx: TransactionEvent = new TestTransactionEvent()
+      .addTraces({
+        to: EXTERNAL_ADDR_1,
+        from: AUGUSTUS_ADDR,
+        // Manually place the function signature due to function argument complexity
+        input: "0xa94e78ef",
+        output: "0x0",
+      })
+      .addTraces({
+        to: EXTERNAL_ADDR_2,
+        from: AUGUSTUS_ADDR,
+        // Manually place the function signature due to function argument complexity
+        input: "0x46c67b6d",
+        output: "0x0",
+      })
+      .addTraces({
+        to: EXTERNAL_ADDR_1,
+        from: AUGUSTUS_ADDR,
+        // Manually place the function signature due to function argument complexity
+        input: "0xa94e78ef",
+        output: "0x0",
+      })
+      .addTraces({
+        to: EXTERNAL_ADDR_2,
+        from: AUGUSTUS_ADDR,
+        // Manually place the function signature due to function argument complexity
+        input: "0x46c67b6d",
+        output: "0x0",
+      })
```

Comment:
```suggestion
      .addTraces({
        to: EXTERNAL_ADDR_1,
        from: AUGUSTUS_ADDR,
        // Manually place the function signature due to function argument complexity
        input: "0xa94e78ef",
        output: "0x0",
      })
      .addTraces({
        to: EXTERNAL_ADDR_1,
        from: AUGUSTUS_ADDR,
        // Manually place the function signature due to function argument complexity
        input: "0x46c67b6d",
        output: "0x0",
      })
      .addTraces({
        to: EXTERNAL_ADDR_2,
        from: AUGUSTUS_ADDR,
        // Manually place the function signature due to function argument complexity
        input: "0xa94e78ef",
        output: "0x0",
      })
      .addTraces({
        to: EXTERNAL_ADDR_2,
        from: AUGUSTUS_ADDR,
        // Manually place the function signature due to function argument complexity
        input: "0x46c67b6d",
        output: "0x0",
      })
```
Small suggestion to have tried different `to` contracts per function signature, so as to only keep the function signatures the same per set of `Traces`.